run

```diff
@@ -159,7 +189,26 @@ clean() {
 restore() {
   if did restore ; then return 0 ; fi
 
-  paket --silent restore
+  if [ "$USE" = dotnet ] ; then
+    local NEW_CHECKSUM
+    if has md5 ; then
+      SUM=md5
+    elif has md5sum ; then
+      SUM=md5sum
+    else
+      echo "Couldn't find command to compute MD5."
+      return 1
+    fi
+    NEW_CHECKSUM="$(find . -name 'project.json' -print0 | xargs -0 cat | $SUM)"
+    local OLD_CHECKSUM=""
+    if [ -f .dotnet ] ; then OLD_CHECKSUM="$(cat .dotnet)" ; fi
+    if [ "$NEW_CHECKSUM" != "$OLD_CHECKSUM" ] ; then
+      echo -n "$NEW_CHECKSUM" > .dotnet
+      dotnet restore --verbosity Warning
```

Comment:
i tried if the downgrade warning messed with the uptodate check, but is not.

there is no way to speedup `dotnet restore`, because that should be a one time only run (or redo that only if references changes, in that case dotnet build report a warning about not up to date project.lock.json).
if you always run that, it's not possibile, because the `*` ranges `project.json` retry to find a new version each `dotnet restore`

so ignore my comment, good workaround.